src/rules/relayContainerReturnsConnection.ts

```diff
@@ -0,0 +1,54 @@
+import {
+  ASTVisitor,
+  NamedTypeNode,
+  ObjectTypeDefinitionNode,
+  TypeDefinitionNode,
+  ValidationContext,
+} from 'graphql';
+
+import { ValidationError } from '../utils';
+
+export function RelayContainerReturnsConnection(
+  context: ValidationContext,
+): ASTVisitor {
+  const ensureNameDoesNotEndWithContainer = (node: TypeDefinitionNode) => {
+    if (node.name.value.match(/ConnectionContainer$/)) {
+      context.reportError(
+        new ValidationError(
+          'relay-connection-types-spec-pinterest-customization',
+          `Types that end in \`Connection\` must be an object type as per the relay spec. \`${node.name.value}\` is not an object type.`,
+          [node],
+        ),
+      );
+    }
+  };
+
+  return {
+    ScalarTypeDefinition: ensureNameDoesNotEndWithContainer,
+    InterfaceTypeDefinition: ensureNameDoesNotEndWithContainer,
+    UnionTypeDefinition: ensureNameDoesNotEndWithContainer,
+    EnumTypeDefinition: ensureNameDoesNotEndWithContainer,
+    InputObjectTypeDefinition: ensureNameDoesNotEndWithContainer,
+    ObjectTypeDefinition(node: ObjectTypeDefinitionNode) {
+      const typeName = node.name.value;
+      if (!typeName.endsWith('Container')) {
```

Comment:
I think we want to check that `!typeName.endsWith('ConnectionContainer')` because we can have other "containers" in the future as well